conda_libmamba_solver/solver.py

```diff
@@ -641,12 +672,39 @@ def _specs_to_tasks_remove(self, in_state: SolverInputState, out_state: SolverOu
             self._check_spec_compat(spec)
             tasks[key].append(str(spec))
 
-        return tasks
+        return dict(tasks)
 
-    def _problems_to_specs(self, problems: str, previous: Mapping[str, MatchSpec]):
-        if self.solver is None:
-            raise RuntimeError("Solver is not initialized. Call `._setup_solver()` first.")
+    def _specs_to_tasks_conda_build(
+        self, in_state: SolverInputState, out_state: SolverOutputState
+    ):
+        only_dot_or_digit_re = re.compile(r"^[\d\.]+$")
+
+        def fix_version_field(spec: MatchSpec):
+            """Fix taken from mambabuild"""
+            if spec.version:
+                version_str = str(spec.version)
+                if re.match(only_dot_or_digit_re, version_str):
+                    spec_fields = spec.conda_build_form().split()
+                    if version_str.count(".") <= 1:
+                        spec_fields[1] = version_str + ".*"
+                    else:
+                        spec_fields[1] = version_str + "*"
+                    return MatchSpec(" ".join(spec_fields))
+            return spec
```

Comment:
Let's move this out of the function, to reduce the effort needed to define inline functions.